From a0ddf15481e34c58beca616b5ff3f0e9f56256c1 Mon Sep 17 00:00:00 2001
From: Third Eye <dsnthirdeye@gmail.com>
Date: Thu, 14 Nov 2019 10:00:19 +0800
Subject: [PATCH v2] Input: synaptics-rmi4 - add missed operations in remove

The driver forgets to destroy workqueue in remove like what
is done when probe fails.
Add a call to destroy_workqueue to fix it.

Since unregistration will wait for the work to finish, we do
not need to deal with work in remove.

Signed-off-by: Third Eye <dsnthirdeye@gmail.com>
---
Changes in v2:
  - Remove the calls of canceling delay and flushing workqueue.
  - Modify the commit message.

 drivers/input/rmi4/rmi_f54.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/rmi4/rmi_f54.c b/drivers/input/rmi4/rmi_f54.c
index 484ae1f97330..897105b9a98b 100644
--- a/drivers/input/rmi4/rmi_f54.c
+++ b/drivers/input/rmi4/rmi_f54.c
@@ -730,6 +730,7 @@ static void rmi_f54_remove(struct rmi_function *fn)
 
 	video_unregister_device(&f54->vdev);
 	v4l2_device_unregister(&f54->v4l2);
+	destroy_workqueue(f54->workqueue);
 }
 
 struct rmi_function_handler rmi_f54_handler = {
-- 
2.23.0

.0

